fix(curriculum): external cocktail link in CIP (#46347)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md

# --description--

> Cocktail sort works by iterating through lists, comparing adjacent elements and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.

Reviewer: Not necessary, but we could add in a serial comma here:

```suggestion
Cocktail sort works by iterating through lists, comparing adjacent elements, and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.
```

Reviewer: Also not necessary, but we may be able to combine these two sentences:

```suggestion
Cocktail sort works by iterating through lists in both directions, comparing adjacent elements, and swapping them if they’re in the wrong order.
```

> Cocktail sort works by iterating through the list, comparing adjacent elements and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.

Reviewer 2: ```suggestion
Cocktail sort works by iterating through lists, comparing adjacent elements and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.
```

# --description--

> The [cocktail shaker](https://rosettacode.org/wiki/Sorting_algorithms/Cocktail_sort) sort is an improvement on the Bubble Sort. While the data is an `array` of numbers, cocktail sorting them run from both sides from the array. The **firstSide** starts at `arr[0]` and the **secondSide** starts `arr[arr.length - 1]`.
> 
> Cocktail sorting checks if the element `arr[0]` of the **firstSide** is bigger than the one following it, then it swaps their positions, a similar process happens to the **secondSide** it's checking whether the element `arr[arr.length - 1]` is smaller than the one behind it, then swaps their positions. This process is iterated until you get your cocktail sorted array.

Reviewer 3: What do you think of this @Sboonny?

```suggestion
The [cocktail shaker](https://rosettacode.org/wiki/Sorting_algorithms/Cocktail_sort) sort is an improvement on the Bubble Sort. Given an array of numbers, the cocktail shaker sort will traverse the array from start to finish, moving the largest number to the end. Then, traverse the array backwards and move the smallest number to the start. It repeats these two passes, moving the next largest/smallest number to its correct position in the array until it is sorted.
```
